src/components/common/currency-code.vue

```diff
@@ -0,0 +1,32 @@
+<template>
```

Comment:
Well not I think component isn't a good idea, too much useless code. Just add a `computed` variable to CurrenciesStore that will take user currencies array and return an object in format of Record<currencyId, currency>. And then in any place you will need currency.code, you will be able to just make something like `currenciesMap[currencyId].code`